plugins/fireblocks/index.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import '@nomiclabs/hardhat-ethers';
+import fs from 'fs';
+import { extendConfig, extendEnvironment } from 'hardhat/config';
+import { lazyObject } from 'hardhat/plugins';
+import {
+  HardhatConfig,
+  HardhatUserConfig,
+  HttpNetworkConfig,
+} from 'hardhat/types';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import './type-extensions';
+import { Chain } from './from-upstream/chain';
+import { FireblocksSDK } from 'fireblocks-sdk';
+import { FireblocksSigner } from './fireblocks-signer';
+
+type NetworkMap = {
+  [K in CustomHardHatRuntimeEnvironment['network']['name']]: Chain | undefined;
+};
+
+// TODO: move network name translation to our config and make Chain a required config property
```

Comment:
this looks like it might have been completed, but going off memory

Reply:
Not entirely, the names on the left are nori names which need to get factored out to split this into a stand-alone plugin.